File: frontend/src/services/menuService.js
import api from './api';

const menuService = {
  getAll: (params = {}) => api.get('/menus', { params }),
  getById: (id) => api.get(`/menu/${id}`),
  create: (data) => api.post('/menu', data),
  update: (id, data) => api.put(`/menu/${id}`, data),
  delete: (id) => api.delete(`/menu/${id}`),
  getActiveMenus: () => api.get('/menus/active'),
  getUserMenus: () => {
    console.log('Calling getUserMenus API...');
    console.log('Token:', localStorage.getItem('token'));
    return api.get('/menus/me');
  },
};

export default menuService;

File: internal/interfaces/menu/menu_service.go
package interfacemenu

import (
	domainmenu "safety-riding/internal/domain/menu"
	"safety-riding/internal/dto"
	"safety-riding/pkg/filter"
)

type ServiceMenuInterface interface {
	Create(req dto.MenuItemCreate) (domainmenu.MenuItem, error)
	GetByID(id string) (domainmenu.MenuItem, error)
	GetAll(params filter.BaseParams) ([]domainmenu.MenuItem, int64, error)
	GetActiveMenus() ([]domainmenu.MenuItem, error)
	GetUserMenus(userId string) ([]domainmenu.MenuItem, error)
	Update(id string, req dto.MenuItemUpdate) (domainmenu.MenuItem, error)
	Delete(id string) error
}
